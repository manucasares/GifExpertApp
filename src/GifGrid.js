import React from "react";
import { useFetchGifs } from "./hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);


    return (
        <>
            <h3>{category}</h3>

            {loading && <p>Cargando...</p>}

            <div className="gif-grid">
                {images.map(img => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;
