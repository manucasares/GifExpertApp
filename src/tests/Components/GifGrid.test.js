import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../GifGrid";
import "@testing-library/jest-dom";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe("Pruebas en el componente GifGrid", () => {

    test("DeberÃ­a mostrarse correctamente", () => {
        const wrapper = shallow(<GifGrid category="Cartman" />);

        // useFetchGifs.mockReturnValue({
        //     data: [],
        //     loading: true,
        // });

        expect(wrapper).toMatchSnapshot();
    });

    // test("Debe mostrar items cuando se muestras las imagenes useFetchGifs", () => {
        
    //     const gif = [
    //         {
    //             id: "ABC",
    //             url: "https://localhost/cualquier/cosa.jpg",
    //             title: "Cualquier Cosa",
    //         },
    //     ];
        
    //     useFetchGifs.mockReturnValue({
    //         data: gif,
    //         loading: false,
    //     });

    //     const wrapper = shallow(<GifGrid category="Cartman" />);
        
    //     expect(wrapper).toMatchSnapshot();
    // });
});
