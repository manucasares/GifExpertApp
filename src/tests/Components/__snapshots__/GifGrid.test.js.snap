// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Deber√≠a mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Cartman
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="gif-grid"
  />
</Fragment>
`;
